SHOW DATABASES test error

```
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 38.855 s <<< FAILURE! - in org.apache.shardingsphere.test.integration.engine.it.dal.GeneralDALIT
Error:  assertExecute[proxy: db -> MySQL -> Literal -> SHOW DATABASES](org.apache.shardingsphere.test.integration.engine.it.dal.GeneralDALIT)  Time elapsed: 1.753 s  <<< FAILURE!
java.lang.AssertionError: 
Size of actual result set is different with size of expected dat set rows.
Expected: is <1>
     but: was <0>
	at org.apache.shardingsphere.test.integration.engine.it.dal.GeneralDALIT.assertExecuteForStatement(GeneralDALIT.java:75)
	at org.apache.shardingsphere.test.integration.engine.it.dal.GeneralDALIT.assertExecute(GeneralDALIT.java:66)
```

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowDatabasesExecutor.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowDatabasesExecutor.java
@@ -66,7 +66,7 @@ public final class ShowDatabasesExecutor implements DatabaseAdminQueryExecutor {
     
     private boolean checkLikePattern(final String schemaName) {
         Optional<String> pattern = showDatabasesStatement.getLike().map(SQLUtil::getShowLikePattern);
-        return pattern.isPresent() && schemaName.matches(pattern.get());
+        return !pattern.isPresent() || schemaName.matches(pattern.get());
     }
     
     private Collection<ShardingSphereRule> getRules(final String schemaName) {

